### PR TITLE
Implement per-invocation directory cache via DEPS_WORK_DIR

### PR DESCRIPTION
**Pull request by nicoddemus**
As discussed, this opens the door to easily implement caching that is only valid during a deps invocation.

### CODE REVIEW

File: source/python/deps/deps_cli.py [hunk 1]
```diff
@@ -915,85 +917,104 @@ def cli(
     initial_time = time.time()
     global _click_echo_color
     original_auto_wrap_for_ansi = click.utils.auto_wrap_for_ansi
-    try:
```
Comment (nicoddemus) [inline source/python/deps/deps_cli.py:918]: All this code has just been indented.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -1,20 +1,18 @@
-name: build
+name: test
 
 on:
   push:
     branches:
     - "master"
   pull_request:
-  schedule:
```
Comment (nicoddemus) [inline .github/workflows/test.yml:8]: I don't think we need to be running this regularly, besides, if there is no activity in the repository in 90 days scheduled workflows get automatically disabled anyway.

File: source/python/deps/deps_cli.py [hunk 3]
```diff
@@ -915,85 +917,104 @@ def cli(
...
+                command,
+                deps_in_order,
+                required_files_filter=required_files_filter,
+                dry_run=dry_run,
+                verbose=verbose,
+                continue_on_failure=continue_on_failure,
+                here=here,
+                jobs=jobs,
+                jobs_unordered=jobs_unordered,
+            )
```
Comment (prusse-martin) [inline source/python/deps/deps_cli.py:991]: I think you could have put the context just around this function call, narrowing the context.

No command execution (where the foler could be used) happens anywhere in this scope.

**Review by prusse-martin — APPROVED** [review]

File: source/python/deps/deps_cli.py (hunk 3 above)
Comment (nicoddemus) [reply source/python/deps/deps_cli.py:991]: Will do.

**Pull request merged**